dvl-a50/dvlfinder.py

```diff
@@ -0,0 +1,50 @@
+import json
+import nmap3
+from blueoshelper import request
+from typing import Optional
+from loguru import logger
+
+def check_for_proper_dvl(ip: str) -> bool:
+  # Check the API for the product_name field
+  url = f"http://{ip}/api/v1/about"
+  try:
+    return "DVL" in json.loads(request(url))["product_name"]
+  except Exception as e:
+    logger.debug(f"{ip} is not a dvl: {e}")
+    return False
+  json.loads(request(url))
+
+def find_the_dvl() -> Optional[str]:
+    # The dvl always reports 192.168.194.95 on mdns, so we need to take drastic measures.
+    # Nmap to the rescue!
+
+    nmap = nmap3.Nmap()
+    # generate the scan mask from our current ips
+    networks = json.loads(request("http://127.0.0.1/cable-guy/v1.0/ethernet"))
+    current_networks = [network["addresses"] for network in networks]
+    # this looks like [{'ip': '192.168.2.2', 'mode': 'server'}]
+    current_ips = []
+    for network in current_networks:
+      for entry in network:
+        current_ips.append(entry["ip"])
+
+
```

Comment:
usually double spaces between lines in python is used for function or class separation